From c772b42fd6a49bc009c818b04c473596fe58a97c Mon Sep 17 00:00:00 2001
From: Henri Yandell <bayard@apache.org>
Date: Fri, 6 Oct 2006 21:44:55 +0000
Subject: [PATCH] Adding _ and ; to the list of allowed characters within the
 path component of a URL. _ is definitely allowed and ; is allowed but there
 are restrictions on its use and we'll need to refactor UrlValidator to
 properly understand how a ; may be used for path parameterization. This is
 with reference to #VALIDATOR-204.

git-svn-id: https://svn.apache.org/repos/asf/jakarta/commons/proper/validator/trunk@453784 13f79535-47bb-0310-9956-ffa450edef68
---
 src/share/org/apache/commons/validator/UrlValidator.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/share/org/apache/commons/validator/UrlValidator.java b/src/share/org/apache/commons/validator/UrlValidator.java
index a0f34f6ed..8196eb54c 100644
--- a/src/share/org/apache/commons/validator/UrlValidator.java
+++ b/src/share/org/apache/commons/validator/UrlValidator.java
@@ -148,7 +148,7 @@ public class UrlValidator implements Serializable {
      */
     private static final int PARSE_AUTHORITY_EXTRA = 3;
 
-    private static final String PATH_PATTERN = "/^(/[-\\w:@&?=+,.!/~*'%$]*)?$/";
+    private static final String PATH_PATTERN = "/^(/[-\\w:@&?=+,.!/~*'%$_;]*)?$/";
 
     private static final String QUERY_PATTERN = "/^(.*)$/";
 
